### ISSUE TITLE
Is it necessary to rebuild the model every train iteration?

### ISSUE THREAD

**Issue by pangyyyyy** [arghosh/RobustMW-Net#1]
Hi, thanks for the great work! 

I noticed that you rebuild the meta-model every iteration (L129), and I was wondering if that is necessary?

https://github.com/arghosh/RobustMW-Net/blob/cabea1fc98997fc2232c05bdd88a66bf3cfc95dd/trainer.py#L122-L130

Would it have any difference or negative impact if i were to just build the meta-model before the loop and reload the model's `state_dict` every iteration (L130) instead?

**Comment by arghosh** [comment]
Hi. Good catch. I think, it should be the same and faster. But, make sure, gradients are not accumulated. 
This code follows more closely Meta-weight-network implementation.  You can check my other [repo](https://github.com/arghosh/noisy_label_pretrain/blob/5731ea2f3bf99391d68b5eb2f488897ce3b2fdfb/models/noisy_models.py#L348) where I simplified the MWNet implementation using higher package.

**Comment by pangyyyyy** [comment]
@arghosh Thanks for the clarification! [Your implementation using higher package](https://github.com/arghosh/noisy_label_pretrain/blob/5731ea2f3bf99391d68b5eb2f488897ce3b2fdfb/models/noisy_models.py#L348) seems rather neat, does it support distributed training?

**Comment by arghosh** [comment]
My code does not support distributed training. I don't think [higher](https://github.com/facebookresearch/higher) supports data parallel. But, you may pass the meta batch to different GPUs, do local meta step, compute base model gradients in each node; after that, DDP can handle that, I guess.

**Comment by pangyyyyy** [comment]
@arghosh Thanks for the clarification!
